### ISSUE TITLE
Optionally treat errors in console as actual errors, or show them as warnings

### ISSUE THREAD

**Issue by craig-dae** [cypress-io/cypress#6832]
Feature Request: 

Show warnings (or errors) when there are console errors in javascript. 

### Current behavior:

A console error might appear, but the test doesn't know about the error it because the test ran successfully. 

### Desired behavior:

I'd like to know when errors happen silently. If I have some error that didn't break the app, I'd like to know that it happened so I can deal with it. 

### Test code to reproduce

Create an app, and throw a null-reference exception into a non-critical part of code. Run a test that triggers the exception without failing tests.

Our example was an Angular app that had a null reference in the html that didn't affect the part of the app being tested.

**Comment by jennifer-shehane** [comment]
Duplicate of #4933 which is dependent on #4917 (essentially creating and designing 'warning' events in our Test Runner).
